objax/zoo/resnet_v2.py

```diff
@@ -335,6 +367,91 @@ def __init__(self,
+        objax_layer.b = objax.TrainVar(jn.array(keras_layer.__dict__['bias'].numpy()).reshape(bias_shape))
+
+
+def convert_linear_layer(objax_layer, keras_layer):
+    """Convert variables of linear layer from objax to keras."""
+    objax_layer.w = objax.TrainVar(jn.array(keras_layer.__dict__['kernel'].numpy()))
+    if objax_layer.b:
+        bias_shape = objax_layer.b.value.shape
+        objax_layer.b = objax.TrainVar(jn.array(keras_layer.__dict__['bias'].numpy()).reshape(bias_shape))
+
+
+def convert_resblock(objax_block, keras_block):
+    """Convert variables of residual block from objax to keras.
+
+    Name mapping between blocks:
+        proj_conv   : 0_conv (when exists)
+        conv_0      : 1_conv
+        conv_1      : 2_conv
+        conv_2      : 3_conv
+        norm_0      : preact_bn
+        norm_1      : 1_bn
+        norm_2      : 2_bn
+    """
+    if hasattr(objax_block, 'proj_conv'):
+        convert_conv_layer(objax_block.proj_conv, [layer for layer in keras_block if layer.name.endswith('0_conv')][0])
+    convert_conv_layer(objax_block.conv_0, [layer for layer in keras_block if layer.name.endswith('1_conv')][0])
+    convert_conv_layer(objax_block.conv_1, [layer for layer in keras_block if layer.name.endswith('2_conv')][0])
+    convert_conv_layer(objax_block.conv_2, [layer for layer in keras_block if layer.name.endswith('3_conv')][0])
+    convert_bn_layer(objax_block.norm_0, [layer for layer in keras_block if layer.name.endswith('preact_bn')][0])
+    convert_bn_layer(objax_block.norm_1, [layer for layer in keras_block if layer.name.endswith('1_bn')][0])
+    convert_bn_layer(objax_block.norm_2, [layer for layer in keras_block if layer.name.endswith('2_bn')][0])
+
+
+def convert_model(model_keras, model_objax, num_blocks: list, include_top: bool = True):
+    """Convert keras implementation of ResNetV2 into objax."""
+    convert_conv_layer(model_objax[0], model_keras.get_layer('conv1_conv'))
+    for b, j in enumerate(num_blocks):
+        for i in range(j):
+            convert_resblock(model_objax[b + 3][i], [layer for layer in model_keras.layers if layer.name.startswith(
+                'conv{}_block{}'.format(b + 2, i + 1))])
+    convert_bn_layer(model_objax[7], model_keras.get_layer('post_bn'))
+    if include_top:
+        convert_linear_layer(model_objax[10], model_keras.get_layer('predictions'))
+
+
+def load_pretrained_weights_from_keras(arch: str, include_top: bool = True, num_classes: int = 1000):
+    """Function to load weights from keras models.
+    """
+    model_registry = {'ResNet50': {'num_blocks': [3, 4, 6, 3]},
+                      'ResNet101': {'num_blocks': [3, 4, 23, 3]},
+                      'ResNet152': {'num_blocks': [3, 8, 36, 3]}}
+    assert tf is not None, 'Please install tensorflow dependency to be able to load keras weights.'
+    assert arch in model_registry, f'Model weights does not exist for {arch}.'
+    if num_classes != 1000:
```

Comment:
How about we remove this and add the following
```
include_top = include_top and num_classes == 1000
```
When num_classes != 1000, include_top should be set False. I thought this may be better to be auto-corrected since user's intention is clear when using num_classes to be different from 1000. However, it also make sense to throw an error to warn user if configurations are not compatible though. WDYT?